pkg/azure/resources.go

```diff
@@ -139,6 +143,26 @@ func GetResourceInfoFromResourceDesc(ctx context.Context, resource *paragliderpb
 	return handler.getResourceInfoFromDescription(ctx, resource)
 }
 
+// Verify that resource is supported, and return the resource ID
+func GetResourceIDFromName(ctx context.Context, handler *AzureSDKHandler, resourceGroup string, resourceName string) (string, error) {
```

Comment:
Based on the name, I would expect this to just do what `getVmUri` does. Maybe rename to something like `ConfirmResourceExists`